providers/implementations/keymgmt/ec_kmgmt.c

```diff
@@ -730,6 +731,35 @@ int ec_set_params(void *key, const OSSL_PARAM params[])
     return 1;
 }
 
+static
+int ec_validate(void *keydata, int selection)
+{
+    EC_KEY *eck = keydata;
+    int ok = 0;
+    BN_CTX *ctx = BN_CTX_new_ex(ec_key_get_libctx(eck));
+
+    if (ctx == NULL)
+        return 0;
+
+    if ((selection & EC_POSSIBLE_SELECTIONS) != 0)
+        ok = 1;
+
+    if ((selection & OSSL_KEYMGMT_SELECT_DOMAIN_PARAMETERS) != 0)
+        ok = ok && EC_GROUP_check(EC_KEY_get0_group(eck), ctx);
+
+    if ((selection & OSSL_KEYMGMT_SELECT_PUBLIC_KEY) != 0)
+        ok = ok && ec_key_public_check(eck, ctx);
+
+    if ((selection & OSSL_KEYMGMT_SELECT_PRIVATE_KEY) != 0)
+        ok = ok && ec_key_private_check(eck);
+
+    if ((selection & OSSL_KEYMGMT_SELECT_KEYPAIR) == OSSL_KEYMGMT_SELECT_KEYPAIR)
+        ok = ok && ec_key_pairwise_check(eck, ctx);
```

Comment:
When calling `ec_key_pairwise_check()`, is it *necessary* to call `ec_key_public_check()` and `ec_key_private_check()` first, or do you end up doing double work in that case?

Reply:
may be a tiny bit of overlap - but they are completely different tests.

Reply:
Ok then